atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OracleDdlTable.java

```diff
@@ -118,6 +117,32 @@ public void create(byte[] tableMetadata) {
                 "INSERT INTO " + config.metadataTable().getQualifiedName() + " (table_name, table_size) VALUES (?, ?)");
     }
 
+    private void maybeAlterTableToHaveOverflowColumn() {
+        if (config.alterTablesOrMetadataToMatch().contains(tableRef)) {
+            try {
+                String shortTableName = oracleTableNameGetter.getInternalShortTableName(conns, tableRef);
+                if (overflowTableHasMigrated() && overflowTableExists() && !overflowColumnExists(shortTableName)) {
+                    alterTableToHaveOverflowColumn(shortTableName);
+                }
+            } catch (TableMappingNotFoundException e) {
+                Throwables.rewrapAndThrowUncheckedException(
+                        "Unable to alter table to have overflow column due to a table mapping error.", e);
+            }
+        }
+    }
+
+    private void alterTableToHaveOverflowColumn(String shortTableName) {
+        log.info(
+                "Altering table to have overflow column to match metadata.",
+                UnsafeArg.of("tableReference", tableRef),
+                UnsafeArg.of("shortTableName", shortTableName));
+        conns.get().executeUnregisteredQuery("ALTER TABLE " + shortTableName + " ADD (overflow NUMBER(38))");
```

Comment:
One thing I realised here is that `ALTER TABLE` is not idempotent, so we have a small race condition (if multiple clients verify that there is no overflow column, and then one modifies it, the rest will fail with:

`ORA-01430: column being added already exists in table`.

I'd recommend using the executeIgnoringErrors method in this class and ignore on the oracle code `ORA-01430` (and then add tests!)

Reply:
yikes